controllers/dockermachine_controller.go

```diff
@@ -17,20 +17,41 @@ limitations under the License.
 package controllers
 
 import (
+	"bytes"
 	"context"
+	"fmt"
+	"io/ioutil"
+	"time"
 
 	"github.com/go-logr/logr"
+	"github.com/pkg/errors"
+	v1 "k8s.io/api/core/v1"
+	apierrors "k8s.io/apimachinery/pkg/api/errors"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
 	"k8s.io/apimachinery/pkg/runtime/schema"
 	"k8s.io/apimachinery/pkg/types"
 	infrastructurev1alpha2 "sigs.k8s.io/cluster-api-provider-docker/api/v1alpha2"
+	"sigs.k8s.io/cluster-api-provider-docker/kind/actions"
 	capiv1alpha2 "sigs.k8s.io/cluster-api/pkg/apis/cluster/v1alpha2"
 	ctrl "sigs.k8s.io/controller-runtime"
 	"sigs.k8s.io/controller-runtime/pkg/client"
 	"sigs.k8s.io/controller-runtime/pkg/handler"
 	"sigs.k8s.io/controller-runtime/pkg/reconcile"
 	"sigs.k8s.io/controller-runtime/pkg/source"
+	"sigs.k8s.io/kind/pkg/cluster"
+	"sigs.k8s.io/kind/pkg/cluster/constants"
+	"sigs.k8s.io/kind/pkg/cluster/nodes"
 )
 
+const (
+	containerRunningStatus = "running"
+
+	// kind uses 'control-plane' and cluster-api uses 'controlplane'. Both use 'worker'.
+	clusterAPIControlPlaneSetLabel = "controlplane"
```

Comment:
Do you mean cluster-api-provider-aws for `controlplane`? I don't see it in cluster-api

Reply:
This got copied in from some comment I had written a while back...I...think in essence, kind uses `control-plane` as the docker container label and we, at least in CAPA used `controlplane`. This comment might have been a typo or like, something from some ancient YAML. Probably best to update the comment simply stating a label key of `set` with this value indicates a control plane node.